modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIVisibilityByDomainTestCase.java

```diff
@@ -206,44 +217,41 @@ public void testVisibilityForAnotherUserInSameDomainInPublisher() throws APIMana
-
-        List<APIIdentifier> apiPublisherAPIIdentifierList =
-                APIMTestCaseUtils.getAPIIdentifierListFromHttpResponse(
-                        apiPublisherClientAnotherUserOtherDomain.getAllAPIs());
-        assertFalse(APIMTestCaseUtils.isAPIAvailable(apiIdentifier, apiPublisherAPIIdentifierList),
-                "API is  visible to another user in other domain in API Publisher. When Visibility is private." +
-                        getAPIIdentifierString(apiIdentifier));
+    public void testVisibilityForAnotherUserInOtherDomainInPublisher() throws APIManagerIntegrationTestException,
+            ApiException {
+        APIListDTO getAllApisResponse = restAPIPublisherOtherDomainUser.getAllAPIs("carbon.super");
+        assertFalse(APIMTestCaseUtils.isAPIAvailable(apiIdentifier, getAllApisResponse),
+                "API is  visible to another user in other domain in API Publisher. When Visibility is private."
+                        + getAPIIdentifierString(apiIdentifier));
     }
 
 
     @Test(groups = {"wso2.am"}, description = "Test the visibility of API in Store for another user in other domain",
             dependsOnMethods = "testVisibilityForAnotherUserInOtherDomainInPublisher")
     public void testVisibilityForAnotherUserInOtherDomainInStore() throws APIManagerIntegrationTestException {
-        List<APIIdentifier> apiStoreAPIIdentifierList =
-                APIMTestCaseUtils.getAPIIdentifierListFromHttpResponse(apiStoreClientAnotherUserOtherDomain.getAPI());
-        assertFalse(APIMTestCaseUtils.isAPIAvailable(apiIdentifier, apiStoreAPIIdentifierList),
+        org.wso2.am.integration.clients.store.api.v1.dto.APIListDTO apiListDTO =
+                restAPIStoreOtherDomainUser.getAllPublishedAPIs();
+        assertFalse(APIMTestCaseUtils.isAPIAvailableInStore(apiIdentifier, apiListDTO),
                 "API is  visible to another user in other domain in API Store. When Visibility is private. " +
                         getAPIIdentifierString(apiIdentifier));
     }
 
 
     @Test(groups = {"wso2.am"}, description = "Test the visibility of API in Publisher for admin in other domain",
             dependsOnMethods = "testVisibilityForAnotherUserInOtherDomainInStore")
-    public void testVisibilityForAdminInOtherDomainInPublisher() throws APIManagerIntegrationTestException {
-        List<APIIdentifier> apiPublisherAPIIdentifierList =
-                APIMTestCaseUtils.getAPIIdentifierListFromHttpResponse(
-                        apiPublisherClientAdminOtherDomain.getAllAPIs());
-        assertFalse(APIMTestCaseUtils.isAPIAvailable(apiIdentifier, apiPublisherAPIIdentifierList),
+    public void testVisibilityForAdminInOtherDomainInPublisher() throws APIManagerIntegrationTestException,
+            ApiException {
+        APIListDTO getAllApisResponse = restAPIPublisherOtherDomainAdmin.getAllAPIs("carbon.super");
```

Comment:
Let's use the constant

Reply:
Fixed in 09e067607b550f97373afb9510ea71424bed8a67